src/main/app/src/components/valintaperusteet/ValintaperustePage.tsx

```diff
@@ -78,6 +79,15 @@ const ValintaperusteContent = ({
   const painotetutArvosanatVisible =
     hakukohde?.metadata?.hakukohteenLinja?.painotetutArvosanat.length > 0;
 
+  useEffect(() => {
+    const el = window.location.hash
+      ? document.getElementById(window.location.hash.substring(1))
+      : null;
+    if (el) {
+      scrollIntoView(el);
+    }
+  });
```

Comment:
Tämä efekti suoritetaan nyt aina kun komponentti rendataan. Tässä on se riski, että skrollataan jonkin muun muutoksen kuin `location.hash`:n muuttumisen takia. Parempi tapa olisi ajaa useEffect kun `hash`muuttuu. Yhdenmukaisuuden vuoksi voisi myös käyttää `useLocation`-hookkia `react-router-dom`-paketista.

```suggestion
  const { hash } = useLocation();
  useEffect(() => {
    const el = hash
      ? document.getElementById(hash.substring(1))
      : null;
    if (el) {
      scrollIntoView(el);
    }
  }, [hash]);
```